src/ed_py/entity.py

```diff
@@ -0,0 +1,308 @@
+        """
+        entities = [e for e in entities if self.in_volume(VectorStamped.from_framestamped(e.pose), volume_id)]
+
+        return entities
+
+    @property
+    def last_update_time(self):
+        return self._last_update_time
+
+    def distance_to_2d(self, point: kdl.Vector) -> float:
+        """
+        Calculate the distance between this entity's pose and the given point.
+
+        :param point: Assumed to be in the same frame_id as the entity itself
+        :return: the distance between the entity's pose and the point
+
+        >>> pose = kdl.Frame(kdl.Rotation.RPY(1, 0, 0), kdl.Vector(3, 3, 3))
+        >>> e = Entity("dummy", None, "map", pose, None, {}, None, rospy.Time())
+        >>> point = kdl.Vector(1, 1, 1)
+        >>> e.distance_to_2d(point)
+        2.8284271247461903
+        """
+
+        # The length of the difference vector between the pose's position and the point
+        difference = self._pose.p - point
+        difference.z(0)
+        return difference.Norm()
+
+    def distance_to_3d(self, point: kdl.Vector) -> kdl.Vector:
+        """
+        Calculate the distance between this entity's pose and the given point.
+
+        :param point: Assumed to be in the same frame_id as the entity itself
+        :return: the distance between the entity's pose and the point
+
+        >>> pose = kdl.Frame(kdl.Rotation.RPY(1, 0, 0), kdl.Vector(3, 3, 3))
+        >>> e = Entity("dummy", None, "map", pose, None, {}, None, rospy.Time())
+        >>> point = kdl.Vector(1, 1, 1)
+        >>> e.distance_to_3d(point)
+        3.4641016151377544
+        """
+        return (self._pose.p - point).Norm()
+
+    def is_a(self, super_type: str) -> bool:
+        """
+        Check whether the entity is a (subclass of) some supertype
+
+        :param super_type: Representing the name of the super_type
+        :return: True if the entity is a (sub)type of the given super_type
+
+        >>> e = Entity("dummy", "coffee_table", None, None, None, {}, ["coffee_table", "table", "furniture", "thing"], 0)
+        >>> e.is_a("furniture")
+        True
+        >>> e.is_a("food")
+        False
+        """
+        return super_type in self.super_types
+
+    @property
+    def pose(self) -> FrameStamped:
+        """Returns the pose of the Entity as a FrameStamped"""
+        return FrameStamped(self._pose, self.last_update_time, self.frame_id)
+
+    @pose.setter
+    def pose(self, pose):
+        """Setter"""
+        self._pose = tf2_ros.convert(pose, FrameStamped).frame
+
+    @property
+    def person_properties(self) -> Union[PersonProperties, None]:
+        if self._person_properties:
+            return self._person_properties
+        else:
+            rospy.logwarn("{} is not a person".format(self))
+            return None
+
+    @person_properties.setter
+    def person_properties(self, value: PersonProperties):
+        self._person_properties = value
+
+    def __repr__(self):
```

Comment:
missing output type

Reply:
`__repr__` always should return a string.